test/integration/apimachinery/admission_test.go

```diff
@@ -0,0 +1,223 @@
+		}
+		// The AdmissionReview that was sent to the webhook
+		requestedAdmissionReview := v1beta1.AdmissionReview{}
+
+		// The AdmissionReview that will be returned
+		responseAdmissionReview := v1beta1.AdmissionReview{}
+		deserializer := codecs.UniversalDeserializer()
+		if _, _, err := deserializer.Decode(body, nil, &requestedAdmissionReview); err != nil {
+			t.Errorf("Fail to deserialize object: %s with error: %+v", string(body), err)
+			responseAdmissionReview.Response = response(false, err.Error())
+		} else {
+			responseAdmissionReview.Response = response(true, "Admitted by integration testing.")
+		}
+
+		// Return the same UID
+		t.Logf("Admission request for GVK: %+v Operation: %s", requestedAdmissionReview.GroupVersionKind(), requestedAdmissionReview.Request.Operation)
+		responseAdmissionReview.Response.UID = requestedAdmissionReview.Request.UID
+
+		respBytes, err := json.Marshal(responseAdmissionReview)
+		if err != nil {
+			t.Errorf("Marshal of response failed with error: %+v", err)
+
+		}
+		if _, err := w.Write(respBytes); err != nil {
+			t.Errorf("Writing response failed with error: %+v", err)
+		}
+	})
+
+	// Starting Webhook server
+	go webhook.ListenAndServeTLS("", "")
+	webhookEndpoint := "https://127.0.0.1:443/admission-endpoint"
+
+	master := etcd.StartRealMasterOrDie(t, func(opts *options.ServerRunOptions) {
+		// force enable all resources so we can check storage.
+		// TODO: drop these once we stop allowing them to be served.
+		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/deployments"] = "true"
+		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/daemonsets"] = "true"
+		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/replicasets"] = "true"
+		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/podsecuritypolicies"] = "true"
+		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/networkpolicies"] = "true"
+	})
+	defer master.Cleanup()
+
+	if _, err := master.Client.CoreV1().Namespaces().Create(&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testNamespace}}); err != nil {
+		t.Fatal(err)
+	}
+
+	fail := admissionv1beta1.Fail
+	// Attaching Admission webhook to API server
+	_, err = master.Client.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Create(&admissionv1beta1.ValidatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{Name: "integration.test.k8s"},
+		Webhooks: []admissionv1beta1.Webhook{{
+			Name: "integration.test.k8s",
+			ClientConfig: admissionv1beta1.WebhookClientConfig{
+				URL:      &webhookEndpoint,
+				CABundle: localhostCert,
+			},
+			Rules: []admissionv1beta1.RuleWithOperations{{
+				Operations: []admissionv1beta1.OperationType{admissionv1beta1.OperationAll},
+				Rule:       admissionv1beta1.Rule{APIGroups: []string{"*"}, APIVersions: []string{"*"}, Resources: []string{"*/*"}},
+			}},
+			FailurePolicy: &fail,
+		}},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, resources, err := master.Client.Discovery().ServerGroupsAndResources()
```

Comment:
It seems necessary to add handling `err` here.

Reply:
done

Reply:
done